Pause Manager

mmhousto opened the pull request:
Pause Manager has a pause menu, resume button, and main menu button that need to be assigned in inspector.
There is a Pause Menu UI Prefab.
Added Dialogue and Thieves Minigame to build settings to test other exits.

mmhousto committed:
Simple Pause Menu

mmhousto committed:
Added Pause Manager to player prefab. Set 2 exits in castle to Dialogue scene and Thieves Minigame scene for testing.

mmhousto committed:
Update Castle.unity

mmhousto committed:
Pause Menu null check

OscarMacielC commented on Assets/Scripts/Managers/PauseManager.cs:
```diff
@@ -0,0 +1,110 @@
+        }
+    }
+
+    private void Pause()
+    {
```
Pause is not pausing physics, the character stills fall down when paused.
¿Should we add Time.timesale=0 when pausing and 1 to resume?

mmhousto committed:
If scene name contains 'game' or 'Game', then pausing freezes time.

Added Restart method if we wanted to reset the level if a player gets stuck somehow or to restart the minigames.

mmhousto committed:
Update Interact to only be called once until you release and press again.
